pkg/env/env.go

```diff
@@ -343,19 +343,30 @@ func (e *testEnv) Run(m *testing.M) int {
 		}
 	}
 
-	exitCode := m.Run() // exec test suite
-
-	finishes := e.getFinishActions()
-	// attempt to gracefully clean up.
-	// Upon error, log and continue.
-	for _, fin := range finishes {
-		// context passed down to each finish step
-		if e.ctx, err = fin.run(e.ctx, e.cfg); err != nil {
-			klog.V(2).ErrorS(err, "Finish action handlers")
+	defer func() {
```

Comment:
Which one do you think is the better place to handle panic? I added here as this was the only place where the `Finish` steps are being performed. But I am sort of conflicted. Should I move this to `processTests` instead ? cc @vladimirvivien  @ShwethaKumbla

Reply:
As we are adding this feature to handle clean up/run finish task when panics this place looks good to me.

Reply:
Ack.